src/Dmishh/Bundle/SettingsBundle/Tests/CachedManagerTest.php

```diff
@@ -0,0 +1,195 @@
+    }
+
+    public function testAllHit()
+    {
+        $owner = $this->createOwner();
+        $value = array('foo' => 'bar');
+
+        $sm = m::mock('Dmishh\Bundle\SettingsBundle\Manager\SettingsManager');
+
+        $manager = $this->getMockBuilder('Dmishh\Bundle\SettingsBundle\Manager\CachedManager')
+            ->setMethods(array('fetchFromCache', 'storeInCache'))
+            ->setConstructorArgs(array($sm, 4711))
+            ->getMock();
+        $manager->expects($this->once())
+            ->method('fetchFromCache')
+            ->with($this->equalTo(null), $this->equalTo($owner))
+            ->willReturn($value);
+
+        $this->assertEquals($value, $manager->all($owner));
+    }
+
+    public function testSet()
+    {
+        $owner = $this->createOwner();
+        $name = 'name';
+        $value = 'foobar';
+
+        $sm = m::mock('Dmishh\Bundle\SettingsBundle\Manager\SettingsManager');
+        $sm->shouldReceive('set')->once()->with($name, $value, $owner);
+
+        $manager = $this->getMockBuilder('Dmishh\Bundle\SettingsBundle\Manager\CachedManager')
+            ->setMethods(array('invalidateCache'))
+            ->setConstructorArgs(array($sm, 4711))
+            ->getMock();
+        $manager->expects($this->once())
+            ->method('invalidateCache')
+            ->with($this->equalTo($name), $this->equalTo($owner))
+            ->willReturn(null);
+
+        $manager->set($name, $value, $owner);
+    }
+
+    public function testSetMany()
+    {
+        $owner = $this->createOwner();
+        $settings = array('name0' => 'value0', 'name1' => 'value1', 'name2' => 'value2');
+
+        $sm = m::mock('Dmishh\Bundle\SettingsBundle\Manager\SettingsManager');
+        $sm->shouldReceive('setMany')->once()->with($settings, $owner);
+
+        $manager = $this->getMockBuilder('Dmishh\Bundle\SettingsBundle\Manager\CachedManager')
+            ->setMethods(array('invalidateCache'))
+            ->setConstructorArgs(array($sm, 4711))
+            ->getMock();
+        $manager->expects($this->exactly(3))
+            ->method('invalidateCache')
+            ->with($this->logicalOr('name0', 'name1', 'name2'), $owner);
+
+        $manager->setMany($settings, $owner);
+    }
+
+    public function testClear()
+    {
+        $owner = $this->createOwner();
+        $name = 'name';
+
+        $sm = m::mock('Dmishh\Bundle\SettingsBundle\Manager\SettingsManager');
+        $sm->shouldReceive('clear')->once()->with($name, $owner);
+
+        $manager = $this->getMockBuilder('Dmishh\Bundle\SettingsBundle\Manager\CachedManager')
+            ->setMethods(array('invalidateCache'))
+            ->setConstructorArgs(array($sm, 4711))
+            ->getMock();
+        $manager->expects($this->once())
+            ->method('invalidateCache')
+            ->with($this->equalTo($name), $this->equalTo($owner))
+            ->willReturn(null);
+
+        $manager->clear($name, $owner);
+    }
+
+    public function testGetCacheKey()
```

Comment:
I think you can skip testing this protected method, because it covered by other tests and method itself is quite simple. Also, `CachedDummy` can be removed